src/driver-hidpp20.c

```diff
@@ -1250,6 +1275,11 @@ hidpp20drv_read_profile(struct ratbag_profile *profile)
 	if (drv_data->capabilities & HIDPP_CAP_ONBOARD_PROFILES_8100) {
 		hidpp20drv_read_profile_8100(profile);
 		ratbag_profile_set_cap(profile, RATBAG_PROFILE_CAP_DISABLE);
+		unsigned int num_resolutions = ratbag_profile_get_num_resolutions(profile);
```

Comment:
Please declare `num_resolutions` at the top. Also, a blank line before the `for` wouldn't hurt.